### PR TITLE
Fix issue 62 and issue 61

### PR DESCRIPTION
**Pull request by beyse**
Issue #62 reports problems downloading video files from YouTube. 

I think this problem comes from the [youtube_dl](https://docs.rs/youtube_dl/latest/youtube_dl/) crate although I did not prove it. What I have found however is that the `yt-dlp` binary downloaded works just fine. So I decided to invoke it directly in the rust code instead of relying on the code in the youtube_dl crate. 

Issue #61 reports a problem with the documentation. 

So I updated the README.md file with the hint to help avoid this pitfall.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ RUN apt-get install -y ffmpeg
 RUN apt-get install -y build-essential
 RUN apt-get install -y libssl-dev
 RUN apt-get install -y clang libclang-dev
+RUN apt-get install -y yt-dlp
```
Comment (beyse) [inline Dockerfile]: Oops. That is by mistake

File: Dockerfile [hunk 2]
```diff
@@ -20,4 +20,8 @@ ENV PATH="/root/.cargo/bin:${PATH}"
 # Set the working directory 
 WORKDIR /home/Infinite-Storage-Glitch
 
+# Set cargo home to a folder in the working directory this will make rebuild 
+# faster as it allows the cargo cache to be saved between docker runs. 
+ENV CARGO_HOME=/home/Infinite-Storage-Glitch/cargo_home
```
Comment (beyse) [inline Dockerfile:25]: This is optional. I found it useful for me when working with the Docker container. It is not related to any issue however.

**Comment by DvorakDwarf** [comment]
Does this automatically download best quality

**Comment by beyse** [comment]
> Does this automatically download best quality

Short answer: **Yes.**

-------------

Long answer: 

The [code](https://github.com/DvorakDwarf/Infinite-Storage-Glitch/pull/72/files#diff-78dd76f214d1b9c3a9ce479b42959af5639bb9432cc57dd166080a5a9052e4d0R24) I submitted uses `-f mp4` and the yt-dlp [documentation](https://github.com/yt-dlp/yt-dlp#format-selection) states

> By default, yt-dlp tries to download the best available quality if you don't pass any options

and further down:

> You can also use a file extension (currently `3gp`, `aac`, `flv`, `m4a`, `mp3`, `mp4`, `ogg`, `wav`, `webm` are supported) to download the **best quality** format [...] e.g. `-f webm` will download the **best quality** format with the `webm` extension served as a single file.

This strongly suggests that using `-f mp4` will download the best quality format with `mp4` extension.

**Pull request merged**